apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDbImpl.cs

```diff
@@ -83,6 +83,60 @@ public DatastoreDbImpl(string projectId, string namespaceId, DatastoreClient cli
             return new LazyDatastoreQuery(streamer.Sync());
         }
 
+        /// <inheritdoc/>
+        public override AggregationQueryResults RunAggregationQuery(AggregationQuery query, ReadConsistency? readConsistency = null, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                AggregationQuery = query,
+                ProjectId = ProjectId,
+                PartitionId = _partitionId,
+                ReadOptions = GetReadOptions(readConsistency)
+            };
+            return AggregationQueryResults.FromRunAggregationQueryResponse(Client.RunAggregationQuery(request, callSettings));
```

Comment:
I'd be tempted to separate out the RPC from the result conversion (here and in the following method) - because then the sync and async methods will look more obviously the same. (The only difference will be in the Client.RunAggregationQuery line.)

Reply:
(You've already done this in the transaction code, I see.)